src/components/SocialMediaLinks/SocialMediaLinks.tsx

```diff
@@ -0,0 +1,208 @@
+export const socialMediaLinkTypeArray = [
+  "blogs",
+  "facebook",
+  "instagram",
+  "pinterest",
+  "soundcloud",
+  "tiktok",
+  "tumbler",
+  "twitter",
+  "youtube",
+] as const;
+export type SocialMediaLinkType = typeof socialMediaLinkTypeArray[number];
+
+export interface SocialMediaLinkDataProps {
+  /** Optional override for default platform name */
+  labelText?: string;
+  /** Required. Must be one of socialMediaLinkTypeArray */
+  type: SocialMediaLinkType;
+  /** Optional override for default social media url */
+  url?: string;
+}
+
+export interface SocialMediaLinksProps {
+  /** Optional border: straight, circular or none. */
+  borders?: BorderType;
+  /** Optional className you can add in addition to "social-media-links." */
+  className?: string;
+  /** Optional color */
+  color?: ColorType;
+  /** ID that other components can cross-reference for accessibility purposes */
+  id?: string;
+  /** Optional layout */
+  layout?: LayoutTypes;
+  /** Optional array of social media platform types, urls, and label texts */
+  linksData?: SocialMediaLinkDataProps[];
+  /** Optional true/false to display names of platform along with icon
+   *  NOTE: Labels will NOT be shown with a circular border */
+  showLabels?: boolean;
+  /** Optional size: medium, large, xlarge */
+  size?: SizeType;
+}
+
+/* Accepts an array containing one or more data objects, each representing a social media platform and optional
+ * overrides for the label and/or URL.
+ *
+ * @returns an array of data objects for each requested platform type that includes type, iconName, labelText and
+ * url. The labelText and url props will include any values supplied to override the defaults.
+ */
+function getLinksData(platforms: SocialMediaLinkDataProps[]) {
+  let allData = [];
+  platforms.forEach((myPlatform) => {
+    // Get the dataset for this platform.
+    let thisPlatformArray = socialMediaDataMap.filter(
+      (socialMediaSlice) => socialMediaSlice.type === myPlatform.type
+    );
+
+    // The filter function returns an array of objects, so let's just get the objects out of the array.
+    let thisPlatformData = thisPlatformArray[0];
+
+    // If a url value exists in myPlatform use it, else use the default.
+    let newUrl = myPlatform.url ?? thisPlatformData.url;
+
+    // If a labelText value exists in myPlatform use it, else use the default.
+    let newLabelText = myPlatform.labelText ?? thisPlatformData.labelText;
+
+    let thisObj = {
+      type: thisPlatformData.type,
+      iconName: thisPlatformData.iconName,
+      labelText: newLabelText,
+      url: newUrl,
+    };
+    allData.push(thisObj);
+  }); // end foreach
+
+  return allData;
+}
+
+/* Helper function for .mdx <Table... /> component.
+ * It is located in SocialMediaLinks.tsx because I cannot figure out how to create a function in the .mdx file
+ *
+ * @returns the SocialMediaDataMap values as an array of arrays
+ */
+export function GetTableData() {
```

Comment:
We can take a closer look once this is in official PR state.

Reply:
Hmm yeah, you can't really define functions in the .mdx files. One option is to define the entire component to render here, so include this here instead of returning just the data array.

```
return (
    <Table
      columnHeaders={["Type", "Label Text", "URL", "Icon Name"]}
      tableData={tableData}
      showRowDividers={true}
    />
  );
```

Then in the .mdx file you can render it as

```
<Canvas>{getTableData()}</Canvas>
```

It's not ideal because the function is not directly related to the component itself but is rather to show its data prop values. You can always create a new file in this directory just for this function. Or maybe rename `SocialMediaDataMap.ts` as `SocialMediaLinksHelpers.tsx` (or utils?) and add the function there. Just ideas.